### ISSUE TITLE
postTweet posting endlessly in command?

### ISSUE THREAD

**Issue by JoeDawson** [thujohn/twitter#74]
Hello,

I've been using this package for over a year now, this is the only time I've had an issue so first and foremost - thank you.

I've been using `2.0.*@dev` for about a month now as suggested for Laravel 5. I was due to launch a Laravel 5 version of one of my sites this evening, when I hit a bit of a wall when running `composer update` on my production server. Of course, this package had updated since I last did a `composer update` on my local therefore my production was ahead.

So, the previous code I had was:

```php
$twitter = \Twitter::postTweetMedia([
	'status' => twitter_status($photo->female),
	'media[]'=> file_get_contents($photo->path),
	'format' => 'json'
]);
$twitter = json_decode($twitter);
```

This was working perfectly up until about an hour ago when I updated the package locally to fix it here to ensure I can sleep happy.

Following the documentation provided, I've updated my config file (of which contains the new UPLOAD_URL, etc.) and have updated my code to be the following:

```php
$media = \Twitter::uploadMedia([
	'media' => file_get_contents($photo->path)
]);

$twitter = \Twitter::postTweet([
	'status' 	=> twitter_status($photo->female),
	'media_ids' => $media->media_id_string
]);

$twitter = json_decode($twitter);
```

Now, this is working fine - but it has some odd behaviour. At least what I have experienced, the best way I can show you this is suggested you check out this screen recording - I quickly explain what's happening: https://www.youtube.com/watch?v=BhuzrzJL13I

**Note, at the time of posting this - the video is still processing on the above link, but shouldn't take too long.**

So to shorten down what I have done;

- Pulled in latest updates for this package
- Configured as per documentation
- Adjust code to implement new Twitter uploading method
- Code previously was fine (fired once), now continuously fires - doesn't stop until I manually stop the queue worker.

I hope you can help! Thanks, Joe.

**Comment by thujohn** [comment]
Can you see the current job in the queue ?
What does twitter_status() ?

I still didn't try queues so I won't be very helpfull if this is the issue but I read that in the [documentation](http://laravel.com/docs/5.0/queues) :
```
Deleting A Processed Job

Once you have processed a job, it must be deleted from the queue.
If no exception is thrown during the execution of your job,
this will be done automatically.

If you would like to delete or release the job manually,
the Illuminate\Queue\InteractsWithQueue trait
provides access to the queue job release and delete methods.
The release method accepts a single value: the number of seconds
you wish to wait until the job is made available again.
```

**Comment by JoeDawson** [comment]
Hey @thujohn,

I can indeed see the job in the queue, it unfortunately stays in the queue - which is odd. In the video, I briefly show the code for my command, it's supposed to post to Tumblr also (which it did before updating) - but it doesn't even get to posting to Tumblr, and that code is directly after posting to Twitter.

the `twitter_status()` function is a helper I have created for myself, it just generates the Twitter status from the `female` model. (just a string)

With regards to the queue, the job is throwing no exception so should be deleting itself just fine. It's odd that it continually fires the tweet off and doesn't proceed to the code directly below it:

```php
/* ------------------------------------
#. Upload Photo & Post to Twitter
------------------------------------ */
$media = \Twitter::uploadMedia([
	'media' => file_get_contents($photo->path)
]);
$twitter = \Twitter::postTweet([
	'status' 	=> twitter_status($photo->female),
	'media_ids' => $media->media_id_string
]);
$twitter = json_decode($twitter);

/* ------------------------------------
#. Post to Tumblr
------------------------------------ */
$tumblr = new \Tumblr\API\Client(
	getenv('TUMBLR_KEY'), getenv('TUMBLR_SECRET'),
	getenv('TUMBLR_TOKEN'), getenv('TUMBLR_TOKEN_SECRET')
);

// further logic to post to Tumblr
// ...
// persists some data to the database
// ...
// etc.
```

I just find it a little odd that this is continually firing off the Tweet and not proceeding to the code directly below (new instantiation of Tumblr api, posting to Tumblr etc etc.).

Thanks

**Comment by thujohn** [comment]
Very strange…

Try to replace some of your lines by var_dump()
Try to run the twitter's commands directly in a controller

Which version of package have you specified in your composer.json ?

**Comment by JoeDawson** [comment]
It is indeed, I'm currently using `~2.0` of this package.

Will try some tests at lunch and will get back to you.

**Comment by JoeDawson** [comment]
Hey @thujohn, just tested a few things - my error. This line:

```php
$twitter = json_decode($twitter);
```

Was no longer needed, the response was already decoded (at least from what I've tested). Getting another odd bug now though unfortunately - locally, everything is working smoothly.

As I previously mentioned, I had configured everything as per the docs - meaning I had updated the path to the Twitter facade. New path:

```
'Twitter' => 'Thujohn\Twitter\Facades\Twitter'
```

So I assume I'd be ok with production as everything local is fine - but I'm receiving this error now unfortunately:

> production.ERROR: exception 'ErrorException' with message 'Class 'Thujohn\Twitter\TwitterFacade' not found' in /vendor/laravel/framework/src/Illuminate/Foundation/AliasLoader.php:63

I then deleted my vendor directory on my production server, deleted `composer.lock` and run a new `composer install` in the hope it would clear the reference to the old `TwitterFacade`. No luck.

Any ideas why I get that error on my production server, but not locally? 

I would assume the above method (deleting vendor etc.) would've worked.

**Comment by thujohn** [comment]
In fact you don't need json_encode() because you can choose the output format by passing it in parameters. By default it is set to object ( ['format' => 'object|array|json'] )

Maybe you can try ```php artisan clear-compiled``` but not sure because this shouldn't happen.
Have you checked the app.php another time ?

**Comment by JoeDawson** [comment]
I see - thanks @thujohn.

Just tried `php artisan clear-compiled` - no luck unfortunately. The last thing I can think of trying is deleting my site off the server, then installing it again. Will try that this evening.

Thanks for all your help so far.

**Comment by thujohn** [comment]
You can try to search "TwitterFacade" in your files to identify the cause of the problem (composer or laravel)

**Comment by JoeDawson** [comment]
Yeah I tried that - I did a `grep -rl "TwitterFacade" ./` from my site root but only returned the reference in the error logs. :/

**Comment by thujohn** [comment]
And what the debug is saying ?

**Comment by JoeDawson** [comment]
Just as I mentioned previously :(:

> [2015-03-27 14:07:54] production.ERROR: exception 'ErrorException' with message 'Class 'Thujohn\Twitter\TwitterFacade' not found' in /vendor/laravel/framework/src/Illuminate/Foundation/AliasLoader.php:63

**Comment by thujohn** [comment]
No more ?

**Comment by JoeDawson** [comment]
Yep - here's the full stack:

```
[2015-03-27 14:07:54] production.ERROR: exception 'ErrorException' with message 'Class 'Thujohn\Twitter\TwitterFacade' not found' in /vendor/laravel/framework/src/Illuminate/Foundation/AliasLoader.php:63
Stack trace:
#0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'Class 'Thujohn\\...', '/home/forge/ink...', 63, Array)
#1 /vendor/laravel/framework/src/Illuminate/Foundation/AliasLoader.php(63): class_alias('Thujohn\\Twitter...', 'Twitter')
#2 [internal function]: Illuminate\Foundation\AliasLoader->load('Twitter')
#3 /app/Commands/PostPhotoCommand.php(64): spl_autoload_call('Twitter')
#4 [internal function]: InkedFemales\Commands\PostPhotoCommand->handle()
#5 /vendor/laravel/framework/src/Illuminate/Container/Container.php(523): call_user_func_array(Array, Array)
#6 /vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(199): Illuminate\Container\Container->call(Array)
#7 [internal function]: Illuminate\Bus\Dispatcher->Illuminate\Bus\{closure}(Object(InkedFemales\Commands\PostPhotoCommand))
#8 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(141): call_user_func(Object(Closure), Object(InkedFemales\Commands\PostPhotoCommand))
#9 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(InkedFemales\Commands\PostPhotoCommand))
#10 /vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(101): call_user_func(Object(Closure), Object(InkedFemales\Commands\PostPhotoCommand))
#11 /vendor/laravel/framework/src/Illuminate/Bus/Dispatcher.php(209): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#12 /vendor/laravel/framework/src/Illuminate/Queue/CallQueuedHandler.php(42): Illuminate\Bus\Dispatcher->dispatchNow(Object(InkedFemales\Commands\PostPhotoCommand), Object(Closure))
#13 /vendor/laravel/framework/src/Illuminate/Queue/Jobs/Job.php(126): Illuminate\Queue\CallQueuedHandler->call(Object(Illuminate\Queue\Jobs\BeanstalkdJob), Array)
#14 /vendor/laravel/framework/src/Illuminate/Queue/Jobs/BeanstalkdJob.php(51): Illuminate\Queue\Jobs\Job->resolveAndFire(Array)
#15 /vendor/laravel/framework/src/Illuminate/Queue/Worker.php(207): Illuminate\Queue\Jobs\BeanstalkdJob->fire()
#16 /vendor/laravel/framework/src/Illuminate/Queue/Worker.php(159): Illuminate\Queue\Worker->process('beanstalkd', Object(Illuminate\Queue\Jobs\BeanstalkdJob), '3', 0)
#17 /vendor/laravel/framework/src/Illuminate/Queue/Worker.php(113): Illuminate\Queue\Worker->pop('beanstalkd', 'default', 0, '10', '3')
#18 /vendor/laravel/framework/src/Illuminate/Queue/Worker.php(85): Illuminate\Queue\Worker->runNextJobForDaemon('beanstalkd', 'default', 0, '10', '3')
#19 /vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(101): Illuminate\Queue\Worker->daemon('beanstalkd', 'default', 0, 128, '10', '3')
#20 /vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(67): Illuminate\Queue\Console\WorkCommand->runWorker('beanstalkd', 'default', 0, 128, true)
#21 [internal function]: Illuminate\Queue\Console\WorkCommand->fire()
#22 /vendor/laravel/framework/src/Illuminate/Container/Container.php(523): call_user_func_array(Array, Array)
#23 /vendor/laravel/framework/src/Illuminate/Console/Command.php(115): Illuminate\Container\Container->call(Array)
#24 /vendor/symfony/console/Symfony/Component/Console/Command/Command.php(257): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#25 /vendor/laravel/framework/src/Illuminate/Console/Command.php(101): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#26 /vendor/symfony/console/Symfony/Component/Console/Application.php(874): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#27 /vendor/symfony/console/Symfony/Component/Console/Application.php(195): Symfony\Component\Console\Application->doRunCommand(Object(Illuminate\Queue\Console\WorkCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#28 /vendor/symfony/console/Symfony/Component/Console/Application.php(126): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#29 /vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(94): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#30 /artisan(36): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#31 {main} 
```

**Comment by thujohn** [comment]
Can you try a route which returns a new Twitter object ?
```dd(get_class(new \Twitter));```

**Comment by JoeDawson** [comment]
`"Thujohn\Twitter\Facades\Twitter"` on both local and production.

http://inkedfemales.com/test

**Comment by thujohn** [comment]
And you are running the good command ? :)

**Comment by JoeDawson** [comment]
Yep - everything is up to date on production. Just left work, when in in
home I'm going to just delete the site and install again.

Will let you know how that goes :)

**Comment by JoeDawson** [comment]
Hey @thujohn,

I managed to get my site up and working yesterday - thanks for the assistance :)

I have another odd issue where Laravel's `route()` helper is outputting a link to "http://localhost/" rather than my production domain (http://inkedfemales.com). But it only does it from my command.

I created a 'test' route and just did this:

```php
$female = App\Female::first();

return Twitter::postTweet([
	'status' => route('females.show', $female->slug)
]);
```

That works perfectly, on local and production. But in my command, it does exactly the same thing but **doesn't** work in production. Is this related to the Twitter package?
